### PR TITLE
Revamp the FAQ page.

### PR DESCRIPTION
**Pull request by geeknoid**
- The FAQ page now has categories to make the page more scalable.

- Removed the FAQ pages that were added a little while ago and integrate
their content into the main site FAQ page.

- Stop computing sitemap.xml since there's a Jekyll plugin that does that
for us.

Staging site: https://geeknoid.github.io/istio.github.io/faq/

### CODE REVIEW

**Comment by ldemailly** [comment]
very nice

is there a way to make it use anchors, so we could link directly to a section and even better to unfold an answer

(for instance like http://quidscor.sourceforge.net/FAQ.html#proxy )

File: googled6057396539986c3.html [hunk 1]
```diff
@@ -1 +0,0 @@
-google-site-verification: googled6057396539986c3.html
```
Comment (ldemailly) [inline googled6057396539986c3.html]: will this removal affect analytics or something ?

**Comment by ldemailly** [comment]
rake test error (try to run rake test locally before PRs)

- ./_site/faq/index.html
  *  internally linking to sidecar-injection.html#automatic-sidecar-injection, which does not exist (line 2)
     <a href="sidecar-injection.html#automatic-sidecar-injection">prerequisites</a>
  *  trying to find hash of sidecar-injection.html#automatic-sidecar-injection, but /home/circleci/repo/_site/faq/index.html/sidecar-injection.html does not exist (line 2)
     <a href="sidecar-injection.html#automatic-sidecar-injection">prerequisites</a>
- ./_site/troubleshooting/index.html
  *  internally linking to /docs/tasks/telemetry/faq.html#how-do-i-see-all-of-the-configuration-for-mixer-, which does not exist (line 50)
     <a href="/docs/tasks/telemetry/faq.html#how-do-i-see-all-of-the-configuration-for-mixer-">Telemetry FAQ</a>
  *  trying to find hash of /docs/tasks/telemetry/faq.html#how-do-i-see-all-of-the-configuration-for-mixer-, but /home/circleci/repo/_site/docs/tasks/telemetry/faq.html does not exist (line 50)
     <a href="/docs/tasks/telemetry/faq.html#how-do-i-see-all-of-the-configuration-for-mixer-">Telemetry FAQ</a>

**Comment by geeknoid** [comment]
You can already link (for example: https://geeknoid.github.io/istio.github.io/faq/index.html#what-is-istio). It doesn't unfold though, I don't know how to do such a thing. I can research this separately.

That google file in the root was just necessary to authenticate myself as being able to put content on the web site. Now that it's done, Google knows about me by user id and doesn't need this file anymore.

**Comment by geeknoid** [comment]
Bad links have been fixed, but Circle CI is still unhappy but only because of sily timeouts:

- ./_site/blog/istio-service-mesh-for-microservices.html
  *  External link https://www.weave.works/blog/istio-weave-cloud failed: got a time out (response code 301)
- ./_site/docs/setup/kubernetes/quick-start.html
  *  External link https://kubernetes.io/docs/tasks/tools/install-kubectl failed: got a time out (response code 301)

**Pull request merged**